CMakeLists.txt

```diff
@@ -6,48 +6,52 @@
 
 cmake_minimum_required(VERSION 3.3)
 
+# If we are not building as a part of LLVM, build Flang as a
+# standalone project, using LLVM as an external library:
+if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
+  project(Flang)
+endif()
+
 # In order to bootstrap the runtime library we need to skip
 # CMake's Fortran tests
 SET(CMAKE_Fortran_COMPILER_WORKS 1)
-
-if( NOT DEFINED TARGET_ARCHITECTURE )
-  execute_process(COMMAND uname -m OUTPUT_STRIP_TRAILING_WHITESPACE
-                  OUTPUT_VARIABLE TARGET_ARCHITECTURE)
+if (WIN32)
+  SET(CMAKE_Fortran_ABI_COMPILED 0)
+  SET(CMAKE_Fortran_COMPILER_SUPPORTS_F90 1)
```

Comment:
@bryanpkc, both of these are needed. (I accidentally had the second in my command line when I said it's not needed.)
Without the second line it fails with,
```
CMake Error at C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/Platform/Windows-MSVC.cmake:71 (message):
  MSVC compiler version not detected properly:
Call Stack (most recent call first):
  C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/Platform/Windows-Flang-Fortran.cmake:1 (include)
  C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeFortranInformation.cmake:25 (include)
  C:/Users/isuru/projects/flang/build/CMakeFiles/CMakeTmp/CMakeLists.txt:4 (project)


CMake Error at C:/Users/Administrator/miniconda3/envs/llvm/Library/share/cmake-3.20/Modules/CMakeTestFortranCompiler.cmake:70 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  runtime/flang/CMakeLists.txt:7 (enable_language)


-- Configuring incomplete, errors occurred!
```

Reply:
OK, no problem.